spec/index.md

```diff
@@ -2929,7 +2957,7 @@ Finally, we can describe the state of the IC as a record having the following fi
     }
     CanStatus
       = Running
-      | Stopping (List (CallOrigin, Nat))
+      | Stopping (List (CallOrigin, Nat, NoDeadline | Timestamp | Expired))
```

Comment:
Could you please explain what's going on here?

I'm too lazy to figure it out myself, but at first sight it seems to me like when a canister is stopping we're doing or not doing something based on whether the deadline of a call context has expired. I wouldn't expect us to have to do anything under any circumstances when a call context's deadline has expired (except maybe immediately drop any response).

Reply:
This is just a list of stopping calls; each of these calls can have a deadline attached, so we track the deadline. These are basically management canister call contexts, just for this one call.

But generally the main complication in the abstract system in this PR is that once we respond with `SYS_UNKNOWN` to a request, we need to drop any further responses for this request. In the implementation we do this through `CallContextManager::calbacks`, but we don't have this structure in the spec. Instead, we are allowed spooky action at a distance, so we can issue a reject message and simultaneously change in flight messages or downstream call contexts. `Expired` basically means that a `SYS_UNKNOWN` response has already been generated, so while we can proceed with whatever work we're doing, in the end we'll just drop the response (`Expired` feels a bit off as a name, happy to take suggestions)

Reply:
[Damn, was going to reply and figured that pressing *Resolve conversation* would both post and resolve. Instead it just lost my post.]

Thanks a lot for the clarification.

And `Expired` seems quite OK to me.